bootstrap/_bootstrap_common.py

```
@@ -0,0 +1,257 @@
+            sql,
+        ],
+        env_overrides={"PGPASSWORD": db_conf["password"]},
+        capture_output=True,
+    )
+    return (result.stdout or "").strip()
+
+
+def ensure_database_exists(db_conf: Dict[str, str]) -> None:
+    exists = psql_query(
+        db_conf,
+        f"SELECT 1 FROM pg_database WHERE datname = '{db_conf['database']}';",
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)